jumpcloud/resource_user_group.go

```diff
@@ -48,6 +48,16 @@ func resourceUserGroup() *schema.Resource {
 					},
 				},
 			},
+			"members": {
+				Type:             schema.TypeList,
```

Comment:
could use a `schema.TypeSet` here and then wouldn't need the `DiffSuppressFunc`, but because a TypeSet is a map underneath, the keys in state would not be `members.$index`, but instead something like `members.$hashprefix`.

As far as I can see, there's no way to know what that hash is, it's not exposed in the public API of TypeSet, so I wouldn't be able to check the members are correct in the acceptance tests, and I felt that was very important

Reply:
i don't think `members` are supported in the jumpcloud go sdk at the moment unfortunately. https://github.com/TheJumpCloud/jcapi-go/blob/master/v2/user_group_post.go

Reply:
Don't think the `members` is a supported field https://docs.jumpcloud.com/api/2.0/index.html#tag/User-Groups/operation/groups_user_post

Reply:
Assigning members to the jumpcloud group will be using https://registry.terraform.io/providers/cheelim1/jumpcloud/latest/docs/resources/user_group_membership

Reply:
Correct Members is not a supported field for `User-Groups.groups_user_post`. The PR emulates it in the resource by also using the `User-Groups.graph_userGroupMembersPost` api directly to automatically update the group members when creating & updating the group.

The reason I want this is so I can fully manage group membership via terraform.

Using only `user_group_membership` we cannot remove a member from a group that has been added directly in jumpcloud admin interface. First we'd have to import that `jumpcloud_user_group_membership` and that only works if we already know the group members.

We have groups that must only have the members as defined in terraform, and not a mix of terraform and manually added.

Reply:
got it, thanks for explaining

Reply:
thanks for contributing and feel free to give the repo a star ⭐  😄

Reply:
> got it, thanks for explaining

Sorry I didn't explain it better in the PR. It was very late though and I was just happy the tests were finally passing! Next time I'll write a better overview :)

> thanks for contributing and feel free to give the repo a star ⭐ 😄

No problem at all.  The least I can do considering how much we use it!

Reply:
Deployed a new release so you can use it https://github.com/cheelim1/terraform-provider-jumpcloud/releases/tag/v0.3.14

Reply:
Wow that was fast. If there are any issues from this feel free to ping me to look into it :+1: 

Thanks :smile:

Reply:
Should be available in `0.3.17` https://registry.terraform.io/providers/cheelim1/jumpcloud/latest/docs